internal/ethapi/api_extra.go

```diff
@@ -92,3 +92,40 @@ func (s *BlockChainAPI) GetBadBlocks(ctx context.Context) ([]*BadBlockArgs, erro
 	}
 	return results, nil
 }
+
+// isHistoricalStateQueryAllowed returns a nil error only if:
+//   - the node is configured to accept historical proofs queries; or
+//   - the block number given is within the window of recent accepted
+//     blocks and not yet part of the historical blocks.
+//
+// Otherwise, a non-nil error is returned.
+func (s *BlockChainAPI) isHistoricalStateQueryAllowed(blockNumOrHash rpc.BlockNumberOrHash) (err error) {
+	historicalProofs, recentBlocksWindow := s.b.HistoricalConfig()
+	if historicalProofs {
```

Comment:
hmm I was actually under impression that if this returns false we would not even look at the recentBlocksWindows config and just return the error/false so that we limit the API.

Reply:
If `historicalProofs` is false, it means we don't support queries for historical blocks so we need to check the block is within the recent blocks window. Should we then rename `historical-proofs` to for example `historical-state-query-allowed` defaulting to false, to make things clearer?

Reply:
This is replaced now by:

https://github.com/ava-labs/coreth/pull/719/files#diff-7e84219e8a88f0a87eefdad20f859d8f4d6b142c5cba935a193a2904a658c5ecR103-R106

And the query window can only be `0` if the node is running in archive mode and the user specified it to be zero (accept all state queries whatever the block is)

Reply:
This is further changed to:

```go
	queryWindow := uint64(core.TipBufferSize)
	if s.b.IsArchive() {
		queryWindow = s.b.HistoricalStateQueryWindow()
		if queryWindow == 0 {
			return nil
		}
	}
```

I hope this is clearer (hiding previous outdated comment)